rust-runtime/aws-smithy-runtime-api/src/client/orchestrator/error.rs

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use super::BoxError;
+use crate::client::interceptors::context::phase::Phase;
+use crate::client::interceptors::InterceptorError;
+use crate::client::orchestrator::HttpResponse;
+use crate::type_erasure::TypeErasedError;
+use aws_smithy_http::result::{ConnectorError, SdkError};
+use std::fmt::Debug;
+
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum OrchestratorError<E: Debug> {
```

Comment:
@rcoh had a good idea: We can use `SdkError<E, ()>` inside of the orchestrator (could even do something like `type OrchestratorError = SdkError<Error, ()>`), and then map the response on that to a full `SdkError<E, HttpResponse>` when returning from `invoke`.

Reply:
I think that this change would be OK, but I want to hold off on making it until I get to the retry PRs. I think I'm getting the response from the context in those cases, but I can't remember rn.